src/riscv/lib/src/pvm/linux.rs

```diff
@@ -1009,6 +1013,101 @@ impl<M: ManagerBase> SupervisorState<M> {
         // Return 0 as an indicator of success
         Ok(0)
     }
+
+    fn write_rlimit<MC: MemoryConfig>(
+        core: &mut MachineCoreState<impl MemoryConfig, M>,
+        addr: VirtAddr,
+        resource: parameters::Rlimit,
+    ) -> Result<(), Error>
+    where
+        M: ManagerReadWrite,
+    {
+        struct MemorySize<MC>(MC);
+        impl<MC: MemoryConfig> MemorySize<MC> {
+            const SIZE: u64 = MC::TOTAL_BYTES as u64;
+
+            const LOWER_BOUND: () = assert!(MC::TOTAL_BYTES >= u64::MIN as usize);
```

Comment:
What is the purpose of the lower bound? It cannot be false regardless of the value you pick for `TOTAL_BYTES`.